### DISCUSSION TITLE
Sponsoring

### DISCUSSION THREAD

**Discussion by appy-one** [appy-one/acebase#100] [Show and tell]
💯 This is #100 💯!
To celebrate, I've added a new setting for sponsors of the AceBase project...

If you are a sponsor you can set the `sponsor` setting in the AceBase constructor to `true`. 
Doing so will cause the AceBase ASCII art "banner ad" that is logged to the console to disappear:
```
      ___          ______                
     / _ \         | ___ \               
    / /_\ \ ___ ___| |_/ / __ _ ___  ___ 
    |  _  |/ __/ _ \ ___ \/ _` / __|/ _ \
    | | | | (_|  __/ |_/ / (_| \__ \  __/
    \_| |_/\___\___\____/ \__,_|___/\___|
                        realtime database
```

**Why did I do this?**

In a recently created issue, I was asked if there was a way to have the banner hidden. I thought about it for a while and reasoned that by displaying the banner, I was actually advertising AceBase to other developers that might see it in the browser console. Implementing a setting that removes the banner would effectively mean to "remove ads" from the software using AceBase. 

In many mobile apps such options are available by performing an in-app purchase. Ads are a source of income for the developers, and help them to pay their bills and maybe even make a profit so they can spend more time on it. If you don't want those ads, you can have them removed with a quick in-app purchase or subscription. This way, the developer still keeps the revenue and you get rid of the ads.

But this is open source software. So it's free, right?
Yes.
And, no.

I see many OSS developers get frustrated lately about the fact that their work is used in commercial software and being downloaded thousands, or even millions of times a week - without them ever seeing a single cent. I think that is not a good development and will hurt the future of OSS. I don't want to end up in that department so I decided to actively start asking for sponsors.

So if you don't want "the ads", use AceBase commercially, or just really like AceBase - consider sponsoring me, so I don't ever have to feel bad for putting so many hours into its development, support and bugfixing. I'm not asking for truckloads of cash but if you, your company or client makes/saves money with a product using AceBase, I'd really appreciate something in return for my time, efforts and support.

By the way - I might be adding (or moving) features in the future to only work with that flag enabled. Just to bring this to the attention of more people. I won't check if you really are a sponsor, I'll leave that to your integrity!

Thank you all for your interest in the project so far, and a very special thanks to those of you who already have made donations in the past.

You are awesome!
Ewout

[![Sponsor AceBase](https://user-images.githubusercontent.com/26569719/168233053-8e56b243-4140-40ab-9a30-4cb3cc149bfe.svg)](https://github.com/sponsors/appy-one) [![Tweet about AceBase](https://user-images.githubusercontent.com/26569719/168228509-8aea6a61-f228-4742-8c83-572ee02872ab.svg)](https://twitter.com/intent/tweet?button=&url=https://github.com/appy-one/acebase&text=I'm+using+@AcebaseRealtime+in+my+project+to+make+my+life+easier!&button=)

**Comment by dtau00** [comment]
cool, not in a position to offer much yet, but i can at least buy you lunch

**Reply by appy-one** [reply]
Thanks, you rock!

**Comment by donl** [comment]
I'm new here, and really impressed with AceBase.  Thank you for what you have done, created and given away!  I'm looking forward to contributing in what ways I can.

**Comment by clibu** [comment]
Adding a Sponsor option and having features only for sponsors is a good move. I'm currently a sponsor for a couple of OSS libs.

**Comment by yumin-chen** [comment]
Thank you for your hard work creating such an impressive open-source software!

**Comment by prabhu** [comment]
I think a banner is a cool idea!
